frontend/src/hooks/useDataFetcher.ts

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { useAlertOnError } from "./useAlertOnError";
+
+/**
+ * A generic React hook for fetching data asynchronously with TypeScript type inference.
+ *
+ * This hook simplifies API data fetching while providing:
+ * - State management for `data`, `loading`, and `error`.
+ * - Optional data transformation via `transform`.
+ * - Dependency tracking (`deps`) to refetch data when dependencies change.
+ * - Custom error handling with `useAlertOnError`.
+ *
+ * @template T The type of the data returned by the `fetchFunction`.
+ * @template R The transformed type of the data after applying `transform` (defaults to `T`).
+ *
+ * @param fetchFunction - An asynchronous function that returns a Promise resolving to `T`.
+ * @param config - Optional configuration object.
+ * @returns An object containing:
+ *   - `data`: The fetched (and optionally transformed) data.
+ *   - `loading`: A boolean indicating whether the fetch operation is in progress.
+ *   - `error`: A boolean indicating if an error occurred.
+ */
+export function useDataFetcher<T, R = T extends undefined ? never : T>(
```

Comment:
Currently we're missing a lot of `AbortController`'s in data fetchers. Lets take this opportunity to clean this up. I think we can implement it by moving the AbortController instantiation logic to the `useDataFetcher` and have the `fetchFunction` receive and implement it. Usage would be something like:

```
const { data: assignedCoReviewers } = useDataFetcher(
    (signal: AbortSignal) => getDestructionList(uuid, signal),  // Endpoint needs to support `AbortSignal`
    {
      errorMessage:
        "Er is een fout opgetreden bij het ophalen van de vernietigingslijst!",
    },
    [uuid]  // I assumed deps here
  )
```

This WOULD mean that we have to check add a signal param to all existing API methods that don't implement it yet and pass them.